src/directory/api.ts

```diff
@@ -0,0 +1,84 @@
+import { EncoderConfig } from "../file.js"
+import * as UnixFS from "../unixfs.js"
+import type { BlockWriter, WritableBlockStream } from "../file.js"
+export type { Metadata } from "../unixfs.js"
+
+export type {
+  WritableBlockStream,
+  BlockWriter,
+  EncoderConfig,
+  WriterConfig,
+  FileWriter,
+} from "../file/api.js"
+
+export interface DirectoryWriter {
+  // /**
+  //  * Adds new entry (either directory or a file) to this directory. Please note
+  //  * that it is API consumer responsibility to ensure that underlying entry
+  //  * blocks are written into the `BlockQueue` of this `DiretoryWriter` if so
+  //  * desired. To include blocks of the added entry you can use `createFileWriter`
+  //  * and `createDirectoryWriter` methods.
+  //  *
+  //  * ⚠️ Please note that call will throw an exception if the conflicting file
+  //  * name is used.
+  //  *
+  //  * @example
+  //  * ```
+  //  * import as UnixFS from "@ipld/dag-unixfs"
+  //  *
+  //  * const demo = async (writable) => {
+  //  *   const writer = UnixFS.fromWriter(writable.getWriter())
+  //  *   const directory = writer.createDirectory()
+  //  *   const file = writer.createFile()
+  //  *   file.write(new TextEncoder().encode('hello world'))
+  //  *   directory.write('hello.txt',  await file.close())
+  //  *   return await directory.close()
+  //  * })
+  //  *
+  //  * const { readable, writable } = new TransformStream()
+  //  * demo(writable)
+  //  * ```
+  //  */
```

Comment:
is this meant to be commented ?

Reply:
Not double commented, no. Thanks for spotting